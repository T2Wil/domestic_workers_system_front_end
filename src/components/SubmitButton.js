import React from 'react';
import { Button } from 'reactstrap';

function SubmitButton(props) {
  return (
    <Button
      className={props.className}
      onClick={props.onClick}
      data-testid="button"
    >
      {props.label}
    </Button>
  );
}

export default SubmitButton;
