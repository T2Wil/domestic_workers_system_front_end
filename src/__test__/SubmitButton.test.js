import React from 'react';
import ReactDOM from 'react-dom';
import SubmitButton from '../components/SubmitButton';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import ReactTestRenderer from 'react-test-renderer';

afterEach(cleanup);
it('renders without crushing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<SubmitButton />, div);
});

it('should render button correctly', () => {
  const { getByTestId } = render(<SubmitButton id="button" label="log in" />);
  expect(getByTestId('button')).toHaveTextContent('log in');
});
it('should render button correctly', () => {
  const { getByTestId } = render(<SubmitButton id="button" label="log out" />);
  expect(getByTestId('button')).toHaveTextContent('log out');
});

it('matches snapshot', () => {
  const tree = ReactTestRenderer.create(
    <SubmitButton id="button" label="log in" />
  );
  expect(tree).toMatchSnapshot();
});
