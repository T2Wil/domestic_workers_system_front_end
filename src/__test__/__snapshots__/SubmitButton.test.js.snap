// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  aria-label={null}
  className="btn btn-secondary"
  data-testid="button"
  onClick={[Function]}
>
  log in
</button>
`;
